### PR TITLE
Improve jsdocs, Fix onMount arg type

### PR DESCRIPTION
**Pull request by Nagellan**

### CODE REVIEW

File: lifecycle/index.d.ts [hunk 1]
```diff
@@ -87,12 +87,11 @@ export function onNotify<Shared = never, SomeStore extends Store = Store>(
 /**
  * Add listener on first store listener.
  *
+ * We recommend to always use `onMount` instead.
  * See {@link onMount} to add constructor and destructor for the store.
```
Comment (Nagellan) [inline lifecycle/index.d.ts:91]: I haven't noticed this line at the previous PR 😅
Now looks better

**Comment by Nagellan** [comment]
Ok, let's wait until https://github.com/lukeed/tsm/issues/36 is solved.
If maintainer doesn't react, we can just specify NodeJS 18.5.0 version in GitHub workflow config or just make **Test / Node.js Latest Full (pull_request)** check optional, because it tests the code on an **unstable** version of NodeJS which is not cool for a required check imho.

**Comment by ai** [comment]
We do not need to have green CI to merge docs PR

**Comment by Nagellan** [comment]
@ai I updated the version of tsm, so now all checks are green

**Pull request merged**